bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java

```diff
@@ -2860,7 +2881,7 @@ private void addActionSet(MPerspective perspective, MPerspective temporary) {
 		List<String> tags = perspective.getTags();
 		List<String> extendedTags = temporary.getTags();
 		for (String extendedTag : extendedTags) {
-			if (!tags.contains(extendedTag)) {
+			if (extendedTag.startsWith(ModeledPageLayout.ACTION_SET_TAG) && !tags.contains(extendedTag)) {
```

Comment:
why is that needed?

Reply:
This method handles action sets actually. Before my commit, there was only handling for action sets. refer to method extendPerspectives "includeTags".
Since there is also handling for onboarding now, this method need to be restricted now.